### PR TITLE
Add requirements.txt, dynamic spotify detection, and Linux support

### PR DESCRIPTION
**Pull request by TabulateJarl8**
I was a bit bored today and saw your post on Reddit, so I thought that I'd contribute a bit. This pull request adds a few things.

 - `.gitignore` - File for telling git to ignore Python junk like `__pycache__` and also the user's spotipy cache file which contains credentials
 - `requirements.txt` - File containing the project requirements. This makes it easier for the user to install requirements, since they only have to type `pip3 install -r requirements.txt` to install all project requirements.
 - Dynamic spotify path detection. I used `shutil.which()` to automatically detect the spotify file location in the PATH. I still left your path variable, however, so that users can manually override this system.
 - Linux support. I use Linux, so to test the rest of the stuff I had to go ahead and add Linux support. Also, I like things to support more than just Windows in general, so I figured that I'd help out.

I'd definitely recommend testing this before merging it, as I haven't tested this on Windows. Other than that, feel free to ask me any questions.

### CODE REVIEW

**Comment by jobveldhuis** [comment]
You should probably specify the path variable to be None and specifically check for that. The way you currently check it will lead to issues with truthy values. While I cannot imagine a real life problem that will arise, I think its better to be safe, rather than sorry.

File: main.py [hunk 1]
```diff
@@ -33,30 +44,32 @@ def setupSpotifyObject(username, scope, clientID, clientSecret, redirectURI):
...
 
+    restartSpotify(path)
+
     while True:
-        
+
         try:
             current_track = spotify.current_user_playing_track()
-        except:
+        except Exception:
```
Comment (jobveldhuis) [inline main.py:56]: Doesn't the API throw a less broad exception when tokens expire?

File: main.py [hunk 2]
```diff
@@ -33,30 +44,32 @@ def setupSpotifyObject(username, scope, clientID, clientSecret, redirectURI):
...
             pass
-        
+
         time.sleep(1)
 
 if __name__ == '__main__':
-    
-    PATH = ''
+
+    PATH = ""
```
Comment (jobveldhuis) [inline main.py:72]: See my comment on the PR. This should probably be set to None and then checked for is None in the code.

Comment (jobveldhuis) [inline main.py:72]: Also, we could use an environment variable for this? I'm not sure how you feel about that.

**Review by jobveldhuis — CHANGES_REQUESTED** [review]

File: main.py (hunk 1 above)
Comment (TabulateJarl8) [reply main.py:56]: Does it throw a `spotipy.exceptions.SpotifyException`? I'll specify that, but I couldn't ever get that exception to be thrown even when I removed the application

File: main.py (hunk 2 above)
Comment (TabulateJarl8) [reply main.py:72]: We could, would you like me to add that really quick?

File: main.py (hunk 1 above)
Comment (TabulateJarl8) [reply main.py:56]: I just checked the code and I believe it does raise a `spotipy.exceptions.SpotifyException`, which subclasses `Exception`, so either way would probably be fine. I'll go ahead and just use the more specific one

**Comment by loopquantum** [comment]
Sorry if I missed something, I don't do much Linux development, but was anything explicitly necessary to add Linux support? It's just Python it should work out of the box no?

**Comment by Packss** [comment]
> Sorry if I missed something, I don't do much Linux development, but was anything explicitly necessary to add Linux support? It's just Python it should work out of the box no?

Python does work out of the box but commands on linux are different, "taskkill" only works for windows.

Besides that, would "which" actually work on windows? idk if spotify is included on the PATH env.

**Comment by TabulateJarl8** [comment]
> Besides that, would "which" actually work on windows? idk if spotify is included on the PATH env.

We could always raise an error if PATH is `None` and `shutil.which('spotify')` is `None` saying that spotify wasn't detected. That would be one way to go about it

**Comment by daspartho** [comment]
Integerated in #10 
Anyways thanks for contributing =)

**Pull request closed without merging**